### PR TITLE
Musicgen: CFG is manually added 

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR exists to keep `musicgen`'s current functionalities considering the changes in #24654.

In a nutshell, the #24654 has a more flexible version of CFG (allows negative prompting, is compatible with existing generation methods, doesn't need to expand the batch size by 2 before the fwd pass, lower memory requirements), but would mean an execution time regression on `musicgen` (because it needs 2x forward passes).

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Works for me! Thanks!

**Comment by gante** [comment]
@sanchit-gandhi there is a potential change due to order of operations, depending on the processors commonly used. Would you be able to confirm whether it be okay like this?

**Review by sanchit-gandhi — APPROVED** [review]
Design is good with me and shouldn't affect the logits done in this way! Thanks @gante
